### DISCUSSION TITLE
Premultiplied alpha and sRGB

### DISCUSSION THREAD

**Discussion by C-Core** [mikke89/RmlUi#704] [Q&A] [answered]
I'm in the process of upgrading our RmlUi integration to 6.0 and struggle a bit with the premultipied alpha requirement. 
We've have been using linear-space (aka gamma correct) blending for everything so far, means we render to an srgb rendertarget, all our textures are srgb and we convert vertex colors from srgb to linear on upload. This worked fine so far but doesn't work with premultiplied alpha since the premultiplication for vertex colors and generated textures is already done in srgb space.
Here is an article that details the problem a bit more: https://hacksoflife.blogspot.com/2022/06/srgb-pre-multiplied-alpha-and.html

We now have a couple of options and I was wondering what the best option would be:
1. Don't do linear-space blending at all (browsers are not doing it, photoshop etc don't do it by default). This would require us to render to an intermediate non-srgb render target which we want to do anyways at some point, so no big deal.
2. Somehow do gamma correct pre-multiplication in RmlUi.
3. Don't do premultiplied alpha in RmlUi itself but let the integration (custom renderer etc) do it.

**Comment by mikke89** [comment, marked as answer]
Oh, good question. Yeah, generally, for our own backends I figured that things look better when not doing gamma correct blending. Maybe it's just because web browsers do it this way, so that's what we're used to. By the way, here's an excellent [stack overflow post](https://stackoverflow.com/a/49321304) comparing different methods for color interpolation. I would say both sRGB and linearized sRGB both have their quirks, I don't think there is a clear winner at least when it comes to that.

So this was part of the reason that I've simply assumed (at least for our backends) that we do everything directly in sRGB. In OpenGL in particular, it's quite simple to just disable the sRGB conversion completely:

https://github.com/mikke89/RmlUi/blob/7a8aceafcc501c5b09ea819bed96d273d9382749/Backends/RmlUi_Renderer_GL3.cpp#L872-L873

So I would say just stick to conventions here, and do number 1.

**Reply by C-Core** [reply]
Thanks for the fast reply, I also had a slight tendency to go with option 1 so I'll go ahead and implement that.
